NPB3.3.1/NPB3.3-OMP/common/hooks.c

```diff
@@ -1,11 +1,15 @@
 #include <stdio.h>
+#include "m5_mmap.h"
 
 void roi_begin_(){
+
+	// need to mmap /dev/mem
+	map_m5_mem();
```

Comment:
This isn't necessarily the first function that's called. 

I suggest creating a "constructor" function that is called before main(). See https://stackoverflow.com/questions/3081832/can-you-run-a-function-on-initialization-in-c. In this function, you can call `map_m5_mem`

Reply:
pushed a  new commit for this